common/djangoapps/enrollment/api.py

```diff
@@ -225,7 +225,8 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         }
 
     """
-    _validate_course_mode(course_id, mode)
+    if mode is not None:
+        _validate_course_mode(course_id, mode)
```

Comment:
@jimabramson I think present behavior in this regard is unchanged. If `mode` is omitted from the request, the view [continues to use `CourseMode.HONOR` as the default value](https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/views.py#L360).